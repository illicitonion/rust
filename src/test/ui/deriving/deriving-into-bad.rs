// Test that From/Into cannot be derived other than for enums with an int repr.

#![allow(unused)]

#[derive(Into)]
//~^ ERROR `Into` can only be derived for enums with an explicit integer representation [FIXME]
struct Struct {}

#[derive(Into)]
//~^ ERROR `Into` can only be derived for enums with an explicit integer representation [FIXME]
#[repr(C)]
enum NumberC {
    Zero,
    One,
}

#[derive(Into)]
//~^ ERROR `Into` can only be derived for enums with an explicit integer representation [FIXME]
enum NumberNoRepr {
    Zero,
    One,
}

fn main() {}
