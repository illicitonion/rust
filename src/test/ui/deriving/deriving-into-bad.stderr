error[FIXME]: `Into` can only be derived for enums
  --> $DIR/deriving-into-bad.rs:5:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for enums with an explicit integer representation
  --> $DIR/deriving-into-bad.rs:9:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for enums with an explicit integer representation
  --> $DIR/deriving-into-bad.rs:17:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for data-free enums
  --> $DIR/deriving-into-bad.rs:24:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for data-free enums
  --> $DIR/deriving-into-bad.rs:32:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for data-free enums
  --> $DIR/deriving-into-bad.rs:40:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[FIXME]: `Into` can only be derived for data-free enums
  --> $DIR/deriving-into-bad.rs:48:10
   |
LL | #[derive(Into)]
   |          ^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 7 previous errors

